src/components/shared/Fields/Input/HookForm/FormattedInputComponent.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react';
+import { defineMessages } from 'react-intl';
+import Cleave from 'cleave.js/react';
+import {
+  Props as CleaveProps,
+  ReactInstanceWithCleave,
+} from 'cleave.js/react/props';
+import Decimal from 'decimal.js';
+import { useFormContext } from 'react-hook-form';
+
+import Button from '~shared/Button';
+import { HookFormInputProps as InputProps } from './Input';
+
+import styles from '../InputComponent.css';
+
+const MSG = defineMessages({
+  max: {
+    id: `HookForm.Input.FormattedInputComponent.max`,
+    defaultMessage: 'Max',
+  },
+});
+
+interface HookFormFormattedInputComponentProps
+  extends Omit<InputProps, 'placeholder' | 'formattingOptions'>,
+    Omit<CleaveProps, 'onChange' | 'name'> {
+  placeholder?: string;
+}
+
+const setCleaveRawValue = (
+  cleave: ReactInstanceWithCleave,
+  maxAmount: string,
+) => {
+  const decimalValue = new Decimal(maxAmount);
+  if (decimalValue.lt(0.00001) && decimalValue.gt(0)) {
+    cleave.setRawValue(maxAmount);
+  } else {
+    cleave.setRawValue(
+      new Decimal(maxAmount).toDP(5, Decimal.ROUND_DOWN).toString(),
```

Comment:
Why is `5` digits hard coded here ? Shouldn't it be a prop with a default value ?